src/AutoRest.CSharp/MgmtTest/Generation/MgmtExtensionTestWriter.cs

```diff
@@ -0,0 +1,140 @@
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    /// <summary>
+    /// Code writer for resource test.
+    /// </summary>
+    internal class MgmtExtensionTestWriter : MgmtBaseTestWriter
+    {
+        protected MgmtExtensions extensions;
+        protected string TestNamespace => extensions.Type.Namespace + ".Tests.Mock";
+        protected string TypeNameOfThis => extensions.Type.Name + "MockTests";
+
+        protected string TestBaseName => $"MockTestBase";
+
+        public MgmtExtensionTestWriter(CodeWriter writer, MgmtExtensions extensions, IEnumerable<string>? scenarioVariables = default) : base(writer, extensions, scenarioVariables)
+        {
+            this.extensions = extensions;
+        }
+
+        public override void Write()
+        {
+            WriteUsings(_writer);
+
+            using (_writer.Namespace(TestNamespace))
+            {
+                _writer.WriteXmlDocumentationSummary($"Test for {extensions.ResourceName}");
+                _writer.Append($"public partial class {TypeNameOfThis:D} : ");
+                _writer.Line($"{TestBaseName}");
+                using (_writer.Scope())
+                {
+                    WriteTesterCtors();
+                    foreach (var clientOperation in extensions.ClientOperations)
+                    {
+                        _writer.Line();
+                        WriteMethodTest(clientOperation, true, false);
+                    }
+                }
+            }
+        }
+
+        protected void WriteUsings(CodeWriter writer)
+        {
+            writer.UseNamespace("System");
+            writer.UseNamespace("System.Threading.Tasks");
+            writer.UseNamespace("System.Net");
+            writer.UseNamespace("System.Collections.Generic");
+            writer.UseNamespace("Azure.Core.TestFramework");
+            writer.UseNamespace("Azure.ResourceManager.TestFramework");
+        }
+
+        protected void WriteTesterCtors()
+        {
+            _writer.Line();
+            using (_writer.Scope($"public {TypeNameOfThis}(bool isAsync): base(isAsync, RecordedTestMode.Record)"))
+            {
+                _writer.Line($"ServicePointManager.ServerCertificateValidationCallback += (sender, cert, chain, sslPolicyErrors) => true;");
+                _writer.Line($"System.Environment.SetEnvironmentVariable(\"RESOURCE_MANAGER_URL\", $\"https://localhost:8443\");");
+            }
+        }
+
+        protected void WriteMethodTest(MgmtClientOperation clientOperation, bool async, bool isLroOperation)
+        {
+            Debug.Assert(clientOperation != null);
+            var methodName = clientOperation.Name;
+            int exampleIdx = 0;
+            foreach ((var branch, var operation) in GetSortedOperationMappings(clientOperation))
+            {
+                var exampleGroup = MgmtBaseTestWriter.FindExampleGroup(operation);
+                if (exampleGroup is null || exampleGroup.Examples.Count() == 0)
```

Comment:
If `exampleGroup.Examples` is not `IEnumerable` (`ICollection` for example), we do not need the Linq method `Count()`. It should have a `Count` property.
```suggestion
                if (exampleGroup?.Examples.Count == 0)
```